src/main/java/org/tikv/common/util/ChannelFactory.java

```diff
@@ -213,14 +214,15 @@ public ChannelFactory(
     this.keepaliveTimeout = keepaliveTimeout;
     this.idleTimeout = idleTimeout;
     this.certContext = new JksContext(jksKeyPath, jksKeyPassword, jksTrustPath, jksTrustPassword);
+    reloadSslContext();
   }
 
   @VisibleForTesting
   public boolean reloadSslContext() {
     if (certContext != null) {
       SslContextBuilder newBuilder = certContext.reload();
```

Comment:
There is still a race between `isModified` check and the rest of initialization logic.

Reply:
Reload in constructor can help mitigating NPE. But the race can make later certificate reload not working as expected.